tokio-net/src/addr.rs

```diff
@@ -0,0 +1,107 @@
+use tokio_executor::blocking;
+
+use futures_util::future;
+use std::future::Future;
+use std::io;
+use std::net::{IpAddr, SocketAddr};
+
+/// Convert or resolve without blocking to one or more `SocketAddr` values.
+pub trait ToSocketAddrs: sealed::ToSocketAddrsPriv {}
+
+type BoxFuture<T> = Box<dyn Future<Output = io::Result<T>> + Unpin + Send>;
+
+// ===== impl SocketAddr =====
+
+impl ToSocketAddrs for SocketAddr {}
+
+impl sealed::ToSocketAddrsPriv for SocketAddr {
+    type Iter = std::option::IntoIter<SocketAddr>;
+    type Future = BoxFuture<Self::Iter>;
+
+    fn to_socket_addrs(&self) -> Self::Future {
+        let iter = Some(*self).into_iter();
+        Box::new(future::ready(Ok(iter)))
+    }
+}
+
+// ===== impl str =====
+
+impl ToSocketAddrs for str {}
+
+impl sealed::ToSocketAddrsPriv for str {
+    type Iter = std::vec::IntoIter<SocketAddr>;
+    type Future = BoxFuture<Self::Iter>;
+
+    fn to_socket_addrs(&self) -> Self::Future {
+        // First check if the input parses as a socket address
+        let res: Result<SocketAddr, _> = self.parse();
+
+        if let Ok(addr) = res {
+            let iter = vec![addr].into_iter();
+            return Box::new(future::ready(Ok(iter)));
+        }
+
+        // Run DNS lookup on the blocking pool
+        let s = self.to_owned();
+
+        Box::new(blocking::run(move || {
+            std::net::ToSocketAddrs::to_socket_addrs(&s)
+        }))
+    }
+}
+
+// ===== impl (IpAddr, u16) =====
+
+impl ToSocketAddrs for (IpAddr, u16) {}
```

Comment:
Please add impl for `(str, u16)` for parity with std trait (other than that, there re no other important impls in std)